packages/babel-preset/tests/babel-preset.test.ts

```diff
@@ -0,0 +1,101 @@
+import {exec} from 'child_process';
+import {readFileSync, writeFileSync} from 'fs';
+import {resolve as resolvePath} from 'path';
+
+import shopifyCommonPreset from '../index';
+
+describe('babel-preset-e2e-test', () => {
+  describe('legacy decorator tests', () => {
+    it('runs babel successfully when typescript is true', async () => {
+      const baseConfig = configFactory({typescript: true});
+      writeFileSync(
```

Comment:
Does thi require writing config to disk and calling the cli? I think we'd be able to do all this in memory using [babel.transform](https://babeljs.io/docs/en/babel-core/#transform)

```js

import { transform } from "@babel/core";
import shopifyPreset from '../index';

// ...

const babelOptions = {
  presets: [shopifyPreset, {typescript: true}]}
};

const output = transform(`SOME CODE HERE`, babelOptions);
const expectedOutput = `BLAH`;

expect(output).toBe(expectedOutput);
```

Reply:
Throwing in a [`configFile: false`](https://babeljs.io/docs/en/options#configfile) to stop it using the default config file would be useful too